server/sonar-search/src/main/java/org/sonar/search/EsSettings.java

```diff
@@ -158,21 +183,38 @@ private static void configureIndexDefaults(Settings.Builder builder) {
   }
 
   private void configureCluster(Settings.Builder builder) {
-    int replicationFactor = 0;
+    int replicationFactor = props.valueAsInt(ProcessProperties.SEARCH_REPLICAS, 0);
```

Comment:
Replica do not make sense by default, when cluster is disabled. Value should still be 0 by default and be overridable only when cluster is enabled.